### PR TITLE
Feature ci mattermost messages

### PR DESCRIPTION
**Pull request by Davknapp**
Update to use an action that still has a running support. 
Needs approval to use the new action.

- [x] TODO: Don't send a message if none has been constructed (happens after update on a branch)

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [ ] The reviewer executed the new code features at least once and checked the results manually

- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [ ] The code is covered in an existing or new test case using Google Test

#### Github action

- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [ ] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

**Comment by holke** [comment]
I have added the action to the settings. @Davknapp could you please confirm?

**Comment by Davknapp** [comment]
> I have added the action to the settings. @Davknapp could you please confirm?

Can you allow it in the following way:
mattermost/action-mattermost-notify@master
or
mattermost/action-mattermost-notify@*

This is how we use it

**Comment by holke** [comment]
> > I have added the action to the settings. @Davknapp could you please confirm?
> 
> Can you allow it in the following way: mattermost/action-mattermost-notify@master or mattermost/action-mattermost-notify@*
> 
> This is how we use it

Done

File: .github/workflows/mattermost.yml [hunk 1]
```diff
@@ -33,39 +33,36 @@ on:
...
-        with: 
-          username: t8y
-          url: ${{ secrets.MATTERMOST_WEBHOOK_URL }}
-          text: ${{ github.event.pusher.name }} pushed into ${{ github.event.ref }}. See ${{ github.event.compare }} for more details.
-      - name: pull_request
-        uses: Mitigram/gh-action-mattermost@main
+        env:
+          message_build: 1
+        run: |
+          echo "{\"text\":\"github.event.pusher.name }} pushed into ${{ github.event.ref }}. See ${{ github.event.compare }} for more details.\"}" > mattermost.json
```
Comment (holke) [inline .github/workflows/mattermost.yml:55]:
```suggestion
          echo "{\"text\":\"${{ github.event.pusher.name }} pushed into ${{ github.event.ref }}. See ${{ github.event.compare }} for more details.\"}" > mattermost.json
```

**Review by holke — CHANGES_REQUESTED** [review]
Looks good to me.
If you can, some comments in the file would be nice to explain whats going on.

File: .github/workflows/mattermost.yml [hunk 2]
```diff
@@ -33,39 +33,39 @@ on:
...
-#        if: github.event.action == 'synchronize'
-#        with:
-#          username: t8y
-#          url: ${{ secrets.MATTERMOST_WEBHOOK_URL }}
-#          text: Updated Pull-Request ${{ github.event.number }}. See ${{ github.event.pull_request.html_url }} for more details.
+        env:
+          message_build: 1
+        run: |
+          echo "{\"text\":\"Pull request ${{ github.event.number }} has been ${{ github.event.action }}. See ${{ github.event.pull_request.html_url }} for more details.\"}" > mattermost.json
+      # if a message has been written send the message to the mattermost-channel described in the secrets
```
Comment (holke) [inline .github/workflows/mattermost.yml:64]:
```suggestion
      # if a message has been written, send the message to the mattermost-channel described in the secrets
```

**Review by holke — APPROVED** [review]

**Pull request merged**